menas/src/main/scala/za/co/absa/enceladus/menas/services/VersionedModelService.scala

```diff
@@ -285,8 +285,8 @@ abstract class VersionedModelService[C <: VersionedModel with Product with Audit
     }
   }
 
-  private[services] def hasWhitespace(name: String): Boolean = !name.matches("""\w+""")
-  private[services] def hasValidNameChars(name: String): Boolean = name.matches("""[a-zA-Z0-9._-]+""")
+  private[services] def hasWhitespace(name: String): Boolean = name != None.orNull && !name.matches("""\w+""")
```

Comment:
```suggestion
  private[services] def hasWhitespace(name: String): Boolean = Option(name).exists( definedName => !definedName.matches("""\w+"""))
```
and the same below.

Reply:
I like Daniel's suggestion.

Reply:
I thought about this and found it to be too hard to immediately understand what it protects from.

Reply:
In my eyes, your solution's downsides are:
 - `None.orNull` - the reader has to realize this is just `null`, but you are avoiding it for a scala-style warning, even though I'd find it more readable. But you would have to add a scala-style ignore comment, etc.
 - realization of the short-circuiting of the `&&` operator.

Both things are well-established phenomenons but they may require some thought. 

On the other hand, my solution only requires one non-trivial realization that `Option(null)` yields `None`. Again, some thought/experience, but otherwise I find it more easily readable.

Of course, your choice. 😄

Reply:
Fixed